### PR TITLE
[Shader Graph] Update Create Node menu to Seacher

### PR DESCRIPTION
**Pull request by alindmanUnity**
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?

This updates the create node menu to use the same searcher as visual scripting. Supports fuzzy searching, tree view, and keyboard navigation. 
image

---
### Release Notes
Please add any useful notes about the feature/fix that might be helpful for others.

This requires editor `2020.1.0a6` at commit `696a2b1d3b3e ` or newer for valid dependencies. 
To add a node to the graph with the new searcher, you must double-click the entry. This is different from the previous search window but by design. 

---
### Testing status
**Katana Tests**: First off we need to make sure the Katana SRP tests are green?

**Manual Tests**: What did you do?
- [x] Checked new UI names with UX convention
- [x] Tested UI multi-edition + Undo/Redo
- [x] C# and shader warnings (supress shader cache to see them)
- [x] Checked new resources path for the reloader (in devloper mode, you have a button at end of resources that check the pathes)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

Alternative, launch Yamato (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/sg%252Fupdate-searcher

Any test projects to go with this to help reviewers?

---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High? Low

**Halo Effect**: None, Low, Medium, High? Low, adds a dependency to the package, but it is a verified package and a valid version to depend on in trunk 

---
### Comments to reviewers
Notes for the reviewers you have assigned.
QA: original test doc can be found here https://docs.google.com/document/d/1-1xAWL0Pkx_a4BTUPEsySCgPUMt9dT5MRXyFuWSWZ1M/edit?usp=sharing

### CODE REVIEW

**Comment by LandonTownsendUnity** [comment]
Release notes say that it requires editor 2020.1.0a6, but I loaded it up with 19.3.0b4 and it seems to be working fine.

**Comment by LandonTownsendUnity** [comment]
Potentially minor problem: entering "(" into the search bar prints an error to the console

File: com.unity.shadergraph/Editor/Drawing/SearchWindowAdapter.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+using System.Text;
+using UnityEditor.Graphing;
+using UnityEditor.Graphing.Util;
+using UnityEngine;
+using UnityEditor.UIElements;
+using UnityEditor.Experimental.GraphView;
+using UnityEngine.UIElements;
+using UnityEditor.ShaderGraph.Drawing;
+using UnityEditor.Searcher;
+
```
Comment (martint-unity) [inline com.unity.shadergraph/Editor/Drawing/SearchWindowAdapter.cs]: half of those using are not used.

File: com.unity.shadergraph/Editor/Drawing/SearchWindowAdapter.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        {
+            NodeGUID = nodeGUID;
+            
+        }
+
+        
+    }
+    
+}
+
```
Comment (martint-unity) [inline com.unity.shadergraph/Editor/Drawing/SearchWindowAdapter.cs]: This file could use some cleaning. A lot of empty lines :)

**Comment by alindmanUnity** [comment]
searcher package is currently using the obsolete list view API, going to wait and merge after the fix lands in the verified package for trunk

**Pull request merged**